src/app/pages/BuyCrypto/steps/ExchangeStep.tsx

```diff
@@ -0,0 +1,283 @@
+import React, { FC, useEffect, useState } from "react";
+
+import classNames from "clsx";
+
+import CopyButton from "app/atoms/CopyButton";
+import Divider from "app/atoms/Divider";
+import FormSubmitButton from "app/atoms/FormSubmitButton";
+import HashShortView from "app/atoms/HashShortView";
+import { ReactComponent as CopyIcon } from "app/icons/copy.svg";
+import ErrorComponent from "app/pages/BuyCrypto/steps/ErrorComponent";
+import useTopUpUpdate from "app/pages/BuyCrypto/utils/useTopUpUpdate";
+import { exchangeDataInterface } from "lib/exolix-api";
+import { getCurrentLocale, T } from "lib/i18n/react";
+import useCopyToClipboard from "lib/ui/useCopyToClipboard";
+
+interface Props {
+  exchangeData: exchangeDataInterface | null;
+  setExchangeData: (exchangeData: exchangeDataInterface | null) => void;
+  step: number;
+  setStep: (step: number) => void;
+  isError: boolean;
+  setIsError: (error: boolean) => void;
+}
+
+const dateFormatOptions = {
+  day: "numeric",
+  month: "long",
+  year: "numeric",
+  hour: "numeric",
+  minute: "numeric",
+};
+
+const ExchangeStep: FC<Props> = ({
```

Comment:
add typeguard for this step and other steps where it's important to have `exchangeData`